cmd/dagger/terminal.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"sync"
+
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+var terminalMu sync.Mutex
+
+func withTerminal(fn func(stdin io.Reader, stdout, stderr io.Writer) error) error {
+	// only allow one terminal session at a time
+	terminalMu.Lock()
+	defer terminalMu.Unlock()
+
+	if silent || !(progress == "auto" && hasTTY || progress == "tty") {
```

Comment:
@vito I copied this logic over -- any chance we could make terminals work in plain mode?